core/relayer/relayer/relayer.go

```diff
@@ -0,0 +1,197 @@
+		// 相关字段置空
+		block.Transactions = nil
+		block.MerkleTree = nil
+		// failedTxs has been consideration into blockid
+		//block.FailedTxs = nil
+
+		blockBuf, pbErr := proto.Marshal(block)
+		// 序列化数据有问题, 重新获取区块
+		if pbErr != nil {
+			fmt.Println("[relayer] proto.Marshal block failed, try to get again")
+			continue
+		}
+		saveErr := relayer.storage.PutBlockHeader(block.GetBlockid(), blockBuf)
+		if saveErr != nil {
+			fmt.Println("[relayer] put block failed, err:", saveErr)
+			panic("put block into storage error:" + saveErr.Error())
+		}
+		saveErr = relayer.storage.PutHeightBlockid(currHeight, block.GetBlockid())
+		if saveErr != nil {
+			fmt.Println("[relayer] put height blockid failed, err", saveErr)
+			panic("put height blockid error:" + saveErr.Error())
+		}
+		// 更新queryMeta
+		tmpQueryMeta := &relayerpb.QueryMeta{
+			LastQueryBlockHeight: currHeight,
+		}
+		err = relayer.queryBlockCmd.UpdateQueryMeta(tmpQueryMeta)
+		if err != nil {
+			panic("update query meta failed, err:" + err.Error())
+		}
+		relayer.queryMeta = &relayerpb.QueryMeta{
+			LastQueryBlockHeight: currHeight,
+		}
+		currHeight++
+	}
+}
+
+// InitRelayer 初始化relayer
+// 使用一个默认StorageConfig来初始化Storage
+func (relayer *Relayer) InitRelayer(cfg *relay.NodeConfig) {
+	// init storage
+	// 默认配置
+	storePath := "./"
+	fileName := "xuper"
+	defaultStorageConfig := &relay.StorageConfig{
+		StorePath: storePath,
+		FileName:  fileName,
+		KVConfig: &kvdb.KVParameter{
+			DBPath:                filepath.Join(storePath, fileName),
+			KVEngineType:          "default",
+			MemCacheSize:          128,
+			FileHandlersCacheSize: 1024,
+			OtherPaths:            []string{},
+		},
+	}
+	storage, err := relay.NewStorage(defaultStorageConfig)
+	if err != nil {
+		fmt.Println("[relayer] new storage error ", err)
+		return
+	}
+	relayer.storage = storage
+
+	// init queryBlockCmd
+	// 传递SrcChain配置以及存储实例
+	queryBlockCmd := &relay.QueryBlockCommand{
+		Cfg:     cfg.Chains.SrcChain,
+		Storage: storage,
+	}
+	relayer.queryBlockCmd = queryBlockCmd
+	// 初始化rpc链接
+	relayer.queryBlockCmd.InitXchainClient()
+	// 查看是否第一次从原链获取区块头
+	queryMeta, queryMetaErr := relayer.queryBlockCmd.LoadQueryMeta()
+	// 如果为not found, 说明第一次从原链获取区块头
+	if common.NormalizedKVError(queryMetaErr) == common.ErrKVNotFound {
+		relayer.queryMeta = &relayerpb.QueryMeta{
+			LastQueryBlockHeight: cfg.AnchorBlockHeight - 1,
+		}
+	} else if queryMetaErr != nil {
+		panic("load query meta failed, error:" + queryMetaErr.Error())
+	}
+	relayer.queryMeta = &relayerpb.QueryMeta{
+		LastQueryBlockHeight: queryMeta.GetLastQueryBlockHeight(),
+	}
+
+	// init deliverBlockCmd
+	// 传递DstChain配置参数以及Storage实例以及锚点区块高度
+	deliverBlockCmd := &relay.DeliverBlockCommand{
+		Cfg:               cfg.Chains.DstChain,
+		Storage:           storage,
+		AnchorBlockHeight: cfg.AnchorBlockHeight,
+		QueryMeta:         relayer,
+	}
+	relayer.deliverBlockCmd = deliverBlockCmd
+	relayer.deliverBlockCmd.InitXchainClient()
+
+	return
+}
+
+func main() {
```

Comment:
Put the independent binary entry under the `cmd` package, refer to ` xdev`.

Reply:
Corrected.